src/attr_file.c

```diff
@@ -287,6 +288,8 @@ int git_attr_file__parse_buffer(
 
 		if (rule->match.flags & GIT_ATTR_FNMATCH_MACRO) {
 			/* TODO: warning if macro found in file below repo root */
+			if (!allow_macros)
```

Comment:
We cannot remove it from the flags, as otherwise we'd throw errors upon parsing macros in subdirectories. They're not disallowed, but simply ignored instead.

I also hate the new `allow_macros` parameter. But we do not have any filenames available at this point, as the parsing may also work on in-memory buffers without an associated filename :/

Reply:
Wait, that's not entirely true. The issue was that if we have parsed a macro and macros aren't allowed, then we'd treat the macro as if it was a file pattern.